src/fs/mod.rs

```diff
@@ -9,3 +9,10 @@ pub mod feature;
 pub mod fields;
 pub mod filter;
 pub mod mounts;
+
+use lazy_static::lazy_static;
+use std::collections::HashMap;
+use std::sync::Mutex;
+lazy_static! {
+    static ref RECURSIVE_SIZE_HASHMAP: Mutex<HashMap<u64, u64>> = Mutex::new(HashMap::new());
+}
```

Comment:
Might we be able to use a oncelock here instead, since we now have msrv=1.70?

Reply:
I don't know about oncelocks, but looking it up, it says it can only be written to once? Not sure the best way to implement this, or whether it's ideal. The HashMap has k:v pairs added in the recursive method.

Reply:
You'd need a `LazyLock` instead, and those are currently only available in nightly.
https://github.com/rust-lang/rust/issues/109736

Reply:
I implemented your suggested change from #546 and it works fine. To get this compiling, I would have to implement all your other changes. Would you be able to alter this line in your PR after this gets merged so as to keep things organized? I appreciate it :smile:

Reply:
Sure, I'll change it.